machine/machine.go

```diff
@@ -191,6 +194,45 @@ func getNodeIdFromCpuBus(cpuBusPath string, threadId int) (int, error) {
 	return nodeId, nil
 }
 
+/* Look for per-node hugepages info using node id */
+/* Such as: /sys/devices/system/node/node%d/hugepages */
+func getHugePagesInfoFromNode(nodePath string, nodeIndex int) ([]info.HugePagesInfo, error) {
+	hugePagesInfo := []info.HugePagesInfo{}
+	path := filepath.Join(nodePath, fmt.Sprintf("node%d/hugepages", nodeIndex))
+	files, err := ioutil.ReadDir(path)
+	// Ignore if per-node info is not available.
+	if err != nil {
+		klog.Errorf("failed to get hugepages information for node %d: %v", nodeIndex, err)
+		return nil, nil
+	}
+
+	for _, file := range files {
+		fileName := file.Name()
+		pageSize, err := parseCapacity([]byte(fileName), hugePageSizeRegexp)
+		if err != nil {
+			return nil, err
+		}
+
+		file := filepath.Join(path, fileName, "nr_hugepages")
+		num, err := ioutil.ReadFile(file)
+		if err != nil {
+			return nil, err
+		}
+
+		pageNum, err := strconv.ParseUint(string(bytes.TrimSpace(num)), 10, 64)
+		if err != nil {
+			return nil, err
+		}
+
+		hugePagesInfo = append(hugePagesInfo, info.HugePagesInfo{
+			PageSize: pageSize / 1024, // Convert to kB.
```

Comment:
Isn't this size already in `kB`?

A test case would be suitable I guess.

Reply:
In the normal hugepage variant they use the pageSize directly, and since it is parsed from `hugepages-<xyz>kB` that should be correct.

https://github.com/google/cadvisor/blob/f5e7ddf27add8bf6b3ead954d5b09d895a8597d7/machine/info.go#L90

Reply:
it seems we should be able to reuse the common logic for how we parse /sys/mm/kernel/hugepages with what we parse here so we dont do it differently.  can we consolidate on a single approach and reuse what can be reused?

Reply:
Yes, I agree that we should reuse as much as possible. This is handled together in the Kernel, so imo. it makes perfect sense to just implement this once in cadvisor. The only thing that differ between them is the filepath, and maybe a bit different error handling, but the implementation should be similar.

Reply:
@derekwaynecarr @odinuge 
`it seems we should be able to reuse the common logic`
`can we consolidate on a single approach`
`Yes, I agree that we should reuse as much as possible.`
=> I totally agree with it. :)

IMO, I think the below strategy is the best option for us.
1) Move `GetHugePagesInfo()` from `info.go` to `machine.go`.
 _(It means that we will use existing way to parse hugepage capacity, not a regex.)_
2) Then make it(`machine.GetHugePagesInfo()`) to take filepath and to parse capacity from a given file for `reusability`.
_(it will be `machine.GetHugePagesInfo(path string)`)_
3) Now, we can reuse it(`machine.GetHugePagesInfo(path string)`) in both of `info.go` and `machine.go`.
4) In `info.go`, `machine.GetHugePagesInfo(path string)` will be used to parse capacity at the machine level.
5) in 'machine.go', `machine.GetHugePagesInfo(path string)` will be used to parse capacity at the NUMA node level.

How about it?

Reply:
I removed duplicated logic.